app/facades/geocoder_facade.rb

```diff
@@ -2,9 +2,29 @@ class GeocoderFacade
   def self.get_coordinates(address, city, state)
     location = "#{address} #{city},#{state}"
     json_response = GeocoderService.get_coordinates(location)
-    lat_long = {
-      lat: json_response[:results][0][:locations][0][:latLng][:lat],
-      lng: json_response[:results][0][:locations][0][:latLng][:lng]
-    }
+    if GeocoderFacade.validate_location?(address, city, state, json_response)
+      lat_long = {
+        lat: json_response[:results][0][:locations][0][:latLng][:lat],
```

Comment:
I'm trying to think of a way to refactor this without much context into everything else going on.

it's usually, not always, but usually not a good idea to be doing things like `[0]` e.g. using numbered indexes like this. 

you should write some tests to account for any of the following being nil `[:results] , [0], [:locations]` etc... because if they are then this will blow up. 

Basically the question to ask or think about is.. can you trust with 100% your `json_response` will ALWAYs have all those fields and none are omited. 

it's totally fine if they aren't always going to be there or if its possible, you'll just want a test to account for that and make sure this method accounts for it.